src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2310,28 +2321,24 @@ FixedwingPositionControl::Run()
 
 		// handle estimator reset events. we only adjust setpoins for manual modes
 		if (_control_mode.flag_control_manual_enabled) {
-			if (_control_mode.flag_control_altitude_enabled && _local_pos.vz_reset_counter != _alt_reset_counter) {
+			if (_control_mode.flag_control_altitude_enabled && _local_pos.vz_reset_counter != _vz_reset_counter) {
```

Comment:
I guess the intent is to reset when there is an altitude step, not a vertical velocity step, right?
> // make TECS accept step in altitude and demanded altitude

Reply:
yes, I think so too. The naming was messed up. Same story for the _pos_reset_counter.
I would prefer to do it in another PR since this requires some more tests again.